src/libraries/System.Private.CoreLib/src/System/Number.Parsing.cs

```diff
@@ -1999,6 +2069,28 @@ internal static double NumberToDouble(ref NumberBuffer number)
             return number.IsNegative ? -result : result;
         }
 
+        internal static Half NumberToHalf(ref NumberBuffer number)
+        {
+            number.CheckConsistency();
+            Half result;
+
+            if ((number.DigitsCount == 0) || (number.Scale < HalfMinExponent))
+            {
+                result = (Half)0;
```

Comment:
This should probably be `Half.PositiveZero` or `default` since we don't have literal support.